### ISSUE TITLE
ReferenceError on running Login request using newman but passing in postman & python requests

### ISSUE THREAD

**Issue by sivakumarp1988** [postmanlabs/newman#866]
![postman](https://cloud.githubusercontent.com/assets/25072850/21879638/13a1be92-d8c0-11e6-9cb7-13fba107937f.PNG)
![newman](https://cloud.githubusercontent.com/assets/25072850/21879589/bec6aebe-d8bf-11e6-925a-2e43da4dc53d.PNG)

ReferenceError occurred on running Login request using newman but passing in postman & python requests.

C:\json_ms4>newman run precoll.json -g global.json --insecure
newman

Login

→ Login
  POST https://<IPAddress>/<ProductName>/v1.0.0/master/UIController/services/Users/actions/login/invoke [errored]
  2? ReferenceError in test-script

┌─────────────────────────┬──────────┬──────────┐
│                         │ executed │   failed │
├─────────────────────────┼──────────┼──────────┤
│              iterations │        1 │        0 │
├─────────────────────────┼──────────┼──────────┤
│                requests │        1 │        1 │
├─────────────────────────┼──────────┼──────────┤
│            test-scripts │        1 │        1 │
├─────────────────────────┼──────────┼──────────┤
│      prerequest-scripts │        1 │        0 │
├─────────────────────────┼──────────┼──────────┤
│              assertions │        0 │        0 │
├─────────────────────────┴──────────┴──────────┤
│ total run duration: 1797ms                    │
├───────────────────────────────────────────────┤
│ total data received: 0B (approx)              │
├───────────────────────────────────────────────┤
│ average response time: 0ms                    │
└───────────────────────────────────────────────┘

  #  failure

 1.  Error

 2.  ReferenceError

 1. Newman Version (can be found via `newman -v`): 3.3.0
 2. OS details (type, version, and architecture): Windows 2008 R2
 3. Are you using Newman as a library, or via the CLI? CLI
 3. Did you encounter this recently, or has this bug always been there: Recently
 4. Expected behaviour: Login should succeed with status code 200.
 5. Command / script used to run Newman: "newman run precoll.json -g global.json --insecure"
 6. Sample collection, and auxilliary files (minus the sensitive details):
 7. Screenshots (if applicable):

**Comment by kunagpal** [comment]
@sivakumarp1988 Could you also share the Node.js version and the test script snippet where the `ReferenceError` is occurring?

**Comment by sivakumarp1988** [comment]
Node.js version - 7.2.0

Tests:
tests["Login"] = responseCode.code === 200;

**Comment by sivakumarp1988** [comment]
When I export the output to json, I could see below error in json.

"failures": [
      {
        "error": {
          "bytesParsed": 309,
          "code": "HPE_INVALID_CONSTANT",
          "name": "Error",
          "message": "Parse Error",
          "stack": "Error: Parse Error\n    at Error (native)\n    at TLSSocket.socketOnData (_http_client.js:359:20)\n    at emitOne (events.js:96:13)\n    at TLSSocket.emit (events.js:188:7)\n    at readableAddChunk (_stream_readable.js:172:18)\n    at TLSSocket.Readable.push (_stream_readable.js:130:10)\n    at TLSWrap.onread (net.js:542:20)",
          "checksum": "bfd3266c76533d1ba8ebd631d2d94fa5",
          "id": "a8dff224-7f6f-4326-aee6-c25da01c5918",
          "timestamp": 1484547403714,
          "stacktrace": [
            {
              "fileName": null,
              "lineNumber": null,
              "functionName": "Error",
              "typeName": "Object",
              "methodName": null,
              "columnNumber": null,
              "native": true
            },
            {
              "fileName": "_http_client.js",
              "lineNumber": 359,
              "functionName": "TLSSocket.socketOnData",
              "typeName": "TLSSocket",
              "methodName": "socketOnData",
              "columnNumber": 20,
              "native": false
            },
            {
              "fileName": "events.js",
              "lineNumber": 96,
              "functionName": "emitOne",
              "typeName": "Object",
              "methodName": null,
              "columnNumber": 13,
              "native": false
            },
            {
              "fileName": "events.js",
              "lineNumber": 188,
              "functionName": "TLSSocket.emit",
              "typeName": "TLSSocket",
              "methodName": "emit",
              "columnNumber": 7,
              "native": false
            },
            {
              "fileName": "_stream_readable.js",
              "lineNumber": 172,
              "functionName": "readableAddChunk",
              "typeName": "Object",
              "methodName": null,
              "columnNumber": 18,
              "native": false
            },
            {
              "fileName": "_stream_readable.js",
              "lineNumber": 130,
              "functionName": "TLSSocket.Readable.push",
              "typeName": "TLSSocket",
              "methodName": "Readable.push",
              "columnNumber": 10,
              "native": false
            },
            {
              "fileName": "net.js",
              "lineNumber": 542,
              "functionName": "TLSWrap.onread",
              "typeName": "TLSWrap",
              "methodName": "onread",
              "columnNumber": 20,
              "native": false
            }
          ]
        }

**Comment by shamasis** [comment]
That helped a lot. It could be that for some reason the URL being parsed in newman. The issue is with the request URL and not the tests.

Can you determine what is the ultimate URL that is formed in newman?

@czardoz you might need to take over and deep dive.

**Comment by czardoz** [comment]
@sivakumarp1988 This error usually occurs when using HTTPS with HTTP endpoints. The "HPE" means "HTTP Parsing Error". By any chance, are you using a proxy server?

**Comment by sivakumarp1988** [comment]
URL for Login - https://172.18.21.86/XXX/v1.0.0/master/UIController/services/Users/actions/login/invoke

**Comment by sivakumarp1988** [comment]
no proxy is used

**Comment by sivakumarp1988** [comment]
any update on this issue?

**Comment by shamasis** [comment]
Oh God! This has been really long and unattended. Real apologies. We actually missed updating the thread with our findings.

The issue happens when for any reason (they may be many) the response received from server was not parsable. There a number of issues around that and with varied solutions.

We reproduced one such issue in one server that apparently needed a trailing backslash! But for you, it maybe a number of issues - and I am not yet familiar with the technicalities of this.

Closing the issue. If it had not gotten actually fixed, I would need a way to exactly reproduce your server environment (which may be tricky, since this seems to be an internal URL of yours.)
